### PR TITLE
Google auth

### PR DESCRIPTION
**Pull request by evguenirousselot**
In this branch, I worked on enabling google authentication for our app

### CODE REVIEW

File: app/src/main/java/com/github/emilehreich/bootcamp/SignIn.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                if (task.isSuccessful) {
+                    // Sign in success, update UI with the signed-in user's information
+                    val user = auth.currentUser
+                    // Redirect to MainActivity
+                    val intent = Intent(this, MainActivity::class.java)
+                    startActivity(intent)
+                    finish()
+                } else {
+                    // If sign in fails, display a message to the user.
+                    // TODO: Handle sign-in failure
```
Comment (emilehreich) [inline app/src/main/java/com/github/emilehreich/bootcamp/SignIn.kt:59]: Sign in failure still undefined

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        val mainActivitySelector = By.res("com.example.myapplication", "activity_main")
+        device.wait(Until.hasObject(mainActivitySelector), 10000)
+
+        // Verify that the sign-in activity is finished and the main activity is launched
+        intended(hasComponent(hasClassName(MainActivity::class.java.name)))
+
+        Intents.release()
+
+
+        /*
```
Comment (emilehreich) [inline app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt:67]: Commented code not deleted. Is it Buggy?

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        onView(withId(R.id.btn_sign_in)).perform(click())
+
+        // Wait for the email selection dialog to appear
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val emailSelector = By.textContains("evgueni.rousselot@gmail.com")
+        device.wait(Until.hasObject(emailSelector), 10000)
+
+        // Click on the email address that you want to sign in with
+        val email = device.findObject(emailSelector)
+        email.click()
```
Comment (emilehreich) [inline app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt:55]: unhandled null pointer exception

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt (hunk 2 above)
Comment (evguenirousselot) [reply app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt]: All done, I forgot to remove it

File: app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt (hunk 3 above)
Comment (evguenirousselot) [reply app/src/androidTest/java/com/github/emilehreich/bootcamp/SignInToSignOutTest.kt]: THANK YOU SO MUCH, you've found my bug now CI works

File: app/src/main/java/com/github/emilehreich/bootcamp/SignIn.kt (hunk 1 above)
Comment (evguenirousselot) [reply app/src/main/java/com/github/emilehreich/bootcamp/SignIn.kt:59]: the google api already deals with it

**Pull request closed without merging**